### ISSUE TITLE
add a `servo.stop()` function

### ISSUE THREAD

**Issue by macMikey** [arduino-libraries/Servo#127]
think of this as something akin to an e-stop.
i have run into a situation, with the [crunchlabs iturret](https://www.crunchlabs.com/products/ir-turret). at power on, the turret attempts to pitch home. unfortunately, the force of that move can cause the turret to "spike" the magazine (because we don't know the power-on position of the servo, and we can't read it, either).
there are two ways to solve that problem (using this library):
1. implement #71 
2. add `servo.stop()`. this command would act like an e-stop, cutting the signal to the servo, regardless of the `.write()` it is currently executing.
i suppose my code for homing the servo would then look something like this:
```cpp
for ( int i = 0 , i < 5 , i++ ) {
   servo.write (90);
   delay (5);
   servo.stop();
   delay (10);
}
servo.write(90); // just to ensure that we finally make it to 90, if the previous five moves didn't do it
```
in this way, the move to 90 would be broken into five moves (depending on the starting position), reducing the change in angular momentum, and still not requiring us to know the position of the servo.

**Comment by HeavyFalcon678** [comment]
@macMikey to have full stop control, you could connect the 5V pin to a digital pin and set it to high. Whenever you want to stop the servo you could just set that pin low and it would cut power to the servo.

**Comment by edgar-bonet** [comment]
A couple of remarks:

 1. The servo is controlled by a PWM signal, with pulses sent every 20&nbsp;ms. I would then suggest waiting for 20&nbsp;ms between `Servo.write()` and `Servo.stop()`.
 2. Thus functionality can be implemented by issuing `Servo.detach()`, then `Servo.attach()` right before the next `Servo.write()`.

**Comment by macMikey** [comment]
i'm pretty sure i tried to detach the servo, but that didn't solve the problem. it seems that once the servo receives the instruction, it's moving to the instructed position, no matter what.
i also rewrote the servo driver, to see if i could get to it, that way, too. that was also unsuccessful.
i think the only reasonable solution is going to be to switch power to an output pin. i have not tried that, because i got bored, after rewriting the servo driver didn't solve the problem.

**Comment by edgar-bonet** [comment]
@macMikey wrote:
> i'm pretty sure i tried to detach the servo, but that didn't solve the problem. it seems that once the servo receives the instruction, it's moving to the instructed position, no matter what.

This is pretty weird. What Arduino are you using? What type of servo?

The file Servo.h says, about `detach()`: “Stops an attached servos from pulsing its I/O pin”. At least the AVR version (the only one I looked at) seems to do exactly that. Traditional analog servos stop doing anything when their data line is not pulsed.

**Comment by macMikey** [comment]
it may be that the controller is just not fast enough.this issue is with one of the hackpacks, specifically the [ir turret](IR Turret - Infrared Communication - CrunchLabs). when the unit powers on, it attempts to crash-home the pitch servo. unfortunately, the turret's center of mass is far enough away from the magnetic mount that it can detach from the base, either at the start, or at the end of the move.
it appears that once the pulse is sent, the motor won't stop. again, it might be that the detach is not executed until it's already too late.

**Comment by HeavyFalcon678** [comment]
I have looked at this before with my turret, and it looks like the problem is when you use `attach`, not the `write`. There is no way to smoothly attach the servo, because it doesn't know its position before it gets there. When the `attach` method is called, it moves the servo straight to home, and then you can move it as slowly as you like because you know where it is.
